src/account.rs

```diff
@@ -141,7 +163,24 @@ mod tests {
             // init test wallet
             let wallet_path = tmp_folder.join("wallet.json");
             save_dummy_wallet_file(&wallet_path);
-            new_account(&wallet_path, TEST_PASSWORD).unwrap();
+            account::derive_new(&wallet_path, 0, TEST_PASSWORD).unwrap();
+        });
+    }
+
+    #[test]
+    fn derive_account_by_index() {
```

Comment:
Could be good to test with an index that's out of bounds

Reply:
From my understanding, for account derivation, there is theoretically no bound on the number of accounts that can be derived (besides whatever the limit of the integer representation is).